### DISCUSSION TITLE
Apply Alpinejs reactivity to AJAX loaded content

### DISCUSSION THREAD

**Discussion by dionesku** [alpinejs/alpine#1203] [1. Help] [answered]
Hi, 

First of all, I want to say that I really like AlpineJS! Great job! 

Now, I have a scenario for which I couldn't make Alpine work: I'm loading some html content with AJAX calls (say a modal with a form) and would like to be able to have this modal and it's content use the Alpine logic - eg. handle the close button click, sync data between different form fields, handle some validations etc. 
Basically, Alpine should be aware of and initialise newly injected html and handle the logic of the components from that html. 
Is this possible?
Thanks!

**Comment by HugoDF** [comment, marked as answer]
Do you have a sample of it not working?

Alpine watches the DOM for changes and initialises any new Alpine components, so it should work out of the box if you're inserting the HTML into the DOM.

It should also work to use `x-html` to inject your markup.

**Reply by dionesku** [reply]
Thanks for the quick reply! You are right, works as expected. Preparing the sample code to post here, I actually got it working. Not really sure why it seemed to not be working in my previous tests. 
Thanks again!
